### PR TITLE
Handle autoyast failures ending in linuxrc

### PR DESCRIPTION
**Pull request by nadvornik**
Original behavior - there is long timeout and logs are not saved:
https://openqa.opensuse.org/tests/280402

Current behavior:
http://10.100.51.143/tests/13886

It will also need update of needle linuxrc-install-fail.

### CODE REVIEW

File: tests/autoyast/installation.pm [hunk 1]
```diff
@@ -103,6 +116,10 @@ sub run {
                 wait_idle(5);
                 $num_errors++;
             }
+            if (match_has_tag('linuxrc-install-fail')) {
+                save_logs_in_linuxrc("stage1_error$i");
+                die "installation ends in linuxrc";
```
Comment (okurz) [inline tests/autoyast/installation.pm]: that's a better idea than just recording "fail" as it is more obvious in which step the test ends. Can you call the "save_logs_in_linuxrc" in a post fail hook and just do the die here? The post_fail_hook will be called after the die has recorded the test to fail.

Comment (okurz) [inline tests/autoyast/installation.pm]: @nadvornik I am fine if you say this does not make sense to you and we continue as you proposed :-)

Comment (nadvornik) [inline tests/autoyast/installation.pm]: As I said, I don't think we need a hook to handle one specific case.

**Comment by okurz** [comment]
I am ok with the change, please rebase

**Comment by nadvornik** [comment]
ok, rebased, thank you

**Pull request merged**